### PR TITLE
Add support for cpu + mem allocation to vm.change command

### PR DESCRIPTION
**Pull request by dougm**
Refactored resource pool related flags and simulator so we can use the same logic
for VirtualMachine resource allocation.

### CODE REVIEW

**Comment by dougm** [comment]
@embano1 added resource allocation flags to the `vm.change` command, but left `vm.create` as-is for now.  Does that work for your case?

File: govc/flags/resource_allocation_info.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+type ResourceAllocationFlag struct {
+	cpu, mem *types.ResourceAllocationInfo
+}
+
+func NewResourceAllocationFlag(cpu, mem *types.ResourceAllocationInfo) *ResourceAllocationFlag {
+	return &ResourceAllocationFlag{cpu, mem}
+}
+
```
Comment (anfernee) [inline govc/flags/resource_allocation_info.go]: this is just smart.

File: govc/vm/change.go [hunk 2]
```diff
@@ -53,10 +54,38 @@ func init() {
...
+	}
+
+	if r.Limit != nil {
+		return
+	}
+
+	if r.Reservation != nil {
+		return
+	}
+
```
Comment (anfernee) [inline govc/vm/change.go]: do we care `ExpandableReservation` here?

Comment (dougm) [reply govc/vm/change.go]: Good catch.  It seems `ExpandableReservation` cannot be set to `true` for the VM fields.  I'll see about just excluding the `{cpu,mem}.expandable` flags from the vm.change command.

Comment (embano1) [reply govc/vm/change.go]: Yes please exclude, bc it´s a resource pool flag.

**Comment by embano1** [comment]
@dougm 
vm.change would work for now. we would just run it after the VM has been successfully created, e.g. by another provisioning tool (customer uses Foreman).

thx a ton!

File: govc/vm/change.go (hunk 2 above)
Comment (dougm) [reply govc/vm/change.go]: ok done.

**Pull request merged**